### DISCUSSION TITLE
Implementar Plugins/Pontos de contribuição

### DISCUSSION THREAD

**Discussion by tgmti** [AlencarGabriel/ProtheusDoc-VsCode#38] [Ideas]
Implementar pontos de contribuição, semelhante a arquitetura do VsCode. 

Discutirmos uma forma de implementar plugins customizáveis na extensão, para atender casos que fogem ao padrão, como o citado na #37.

Avaliar qual a melhor forma, se através de arquivos de configuração ou extensões separadas.
Talvez implementar parecido com o que foi feito no https://github.com/AlencarGabriel/advpl-switch-project

**Comment by denisgalvani** [comment]
Gostaria de saber como incluir um plugin na extensão. A ideia é para que coloque marcações markdown no texto e estas sejam convertidas em formação HTML na exportação para Documentação HTML. Tornando a saída mais rica, simples e menos poluída no fonte. Como termo entre asteriscos para *itálico*, duplo asterisco para **negrito**, acento grave para `código` e etc.

Fiz algumas tratativas mudando os arquivos gerados, mas a cada nova geração esta alterações são perdidas.

**Comment by AlencarGabriel** [comment]
Olá @denisgalvani ,

ainda não é possível implementar um plugin nesta extensão. Precisamos avaliar primeiro os _contribution points_ que serão interessantes e necessários implementar.

Sua solicitação para permitir que a extensão reconheça a notação Markdown é muito interessante. Em Off alguns já me pediram isso, mas a falta de tempo hoje me impede de implementar. Como você já fez alguma coisa, acho que seria válido você abrir um Pull Request no repositório https://github.com/robsonrosilva/protheusDoc (este quem é responsável pela geração do HTML) com a implementação que você fez. Seria uma ótima contribuição para a comunidade! 

Qualquer coisa só me avisar. Pode também transferir sua issue pra lá, pois lá é onde será feito essas alterações.

**Comment by AlencarGabriel** [comment]
Olá @tgmti e @denisgalvani ,

migrei a issue para o formato discussão (recurso novo do GitHub). 

Acredito ser aqui um lugar melhor para tratarmos esse assunto e retomarmos as discussões e ideias.
